src/NinjaTools.FluentMockServer.API/Models/Logging/LogItem.cs

```diff
@@ -0,0 +1,87 @@
+using System;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Text;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Converters;
+using NinjaTools.FluentMockServer.API.Extensions;
+
+namespace NinjaTools.FluentMockServer.API.Models.Logging
+{
+    [DebuggerDisplay("{DebuggerDisplay(),nq}")]
+    public abstract class LogItem<T> : ILogItem
+    {
+        /// <inheritdoc />
+        [JsonProperty(Order = 0)]
+        public string Id { get; }
+
+        [JsonProperty(Order = 1)]
+        public abstract LogKind Kind { get; }
+
+        [JsonProperty(Order = 2)]
+        public T Content { get; }
+
+
+        public LogItem(string id, T content)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Change the visibility of this constructor to 'protected'.](https://app.codacy.com/manual/git_36/NinjaTools.FluentMockServer/pullRequest?prid=4802642)